python/sglang/srt/layers/layernorm.py

```diff
@@ -260,6 +268,79 @@ def forward_with_allreduce_fusion(
         return self.forward(x, residual)
 
 
+class LayerNorm(CustomOp):
+    def __init__(
+        self,
+        hidden_size: int,
+        eps: float = 1e-6,
+        elementwise_affine: bool = True,
+        bias: bool = True,
+    ) -> None:
+        super().__init__()
+        self.hidden_size = hidden_size
+        self.variance_epsilon = eps
+        self.elementwise_affine = elementwise_affine
+
+        if self.elementwise_affine:
+            self.weight = nn.Parameter(torch.ones(hidden_size, dtype=torch.float32))
+            if bias:
+                self.bias = nn.Parameter(torch.zeros(hidden_size, dtype=torch.float32))
```

Comment:
This forces them to be fp32, which is not correct. Pass dtype as arg to __init__ and use dtype here.

https://github.com/pytorch/pytorch/blob/v2.9.0/torch/nn/modules/normalization.py#L106